Document Mental-Model and Naming choices (#136)
tc39/proposal-temporal: civil.md

# The naming of Civil objects

Famous quote:

    There are only two hard things in Computer Science: cache invalidation and naming things.

    -- Phil Karlton

Infamous quote:

>     Cache invalidation is easy if you don't use a caches.

Reviewer: ```suggestion
    Cache invalidation is easy if you don't use a cache.
```

    -- Phil Dunkel

Which leaves us with a *hard problem in computer science*: the naming of the `Civil` family of objects.

> **There are a set of objects that represents dates, times as well as aggregations and parts thereof which do not contain enough data to tie them to a sepcific timezone or locality: what do we call them?**

Reviewer: ```suggestion
**There are a set of objects that represents dates, times, as well as aggregations and parts thereof which do not contain enough data to tie them to a sepcific timezone or locality: what do we call them?**
```

Before we talk about how to name them, we should be clear on what things we are naming and the [mental model](./mentalmodel.md) they represent:

 * `CivilDate` represens a date without a time or locality
 * `CivilTime` represents a time without a date or timezone or locality
 * `CivilDateTime` represents a date and time without a timezone or locality
 * `CivilYearMonth` repreents a year and month without a day or time or timezone or locality (think: July 2019)
 * `CivilMonthDay` represents a day and month without a year or time or locality (think 25th of December);

**Options:**

 * `Gregorian*` since they are specific to the Gregorian Calendar.
 * `Local*` since they look like what we use to talk about our local time.
 * `Civil*` since they look like what we use in civil life.

## Gregorian

> **Pro:** The reason `Gregorian` was suggested was in view of the fact that we are limiting ourselves to the gregorian calendar. Future implementations of these object could them be named appropriately according to the calendar they implement and be on equal footing to the current set of objects that we intend to support.

Reviewer: ```suggestion
**Pro:** The reason `Gregorian` was suggested was in view of the fact that we are limiting ourselves to the gregorian calendar. Future implementations of these objects could then be named appropriately according to the calendar they implement and be on equal footing to the current set of objects that we intend to support.
```
